reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,145 @@
 )
 from reconcile.status import ExitCodes
 from reconcile.utils import gql
+from reconcile.utils.aws_api import AWSApi
 
 QONTRACT_INTEGRATION = "vpc-peerings-validator"
 
 
+def validate_no_cidr_overlap(
+    query_data: VpcPeeringsValidatorQueryData,
+) -> bool:
+    clusters: list[ClusterV1] = query_data.clusters or []
+
+    for cluster in clusters:
+        peerings_enteries_dict = {}  # type: ignore[var-annotated]
+        if cluster.peering:
+            for peering in cluster.peering.connections:
+                if peering.provider == "account-vpc-mesh":
+                    aws_account_uid = peering.account.uid  # type: ignore[union-attr]
+                    settings = queries.get_secret_reader_settings()
+                    accounts = queries.get_aws_accounts(uid=aws_account_uid)
+                    awsapi = AWSApi(1, accounts, settings=settings, init_users=False)
+                    mesh_results = awsapi.get_vpcs_details(accounts[0])
```

Comment:
We don't need all the vpcs in the target account, we have tags for filtering them. same as here https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/terraform_vpc_peerings.py#L309